Kernel/Net/IPv4Socket.h

```diff
@@ -21,6 +22,8 @@ class NetworkAdapter;
 class TCPPacket;
 class TCPSocket;
 
+using PortAllocationResult = Tuple<KResultOr<u16>, bool>;
```

Comment:
Can we please not use `Tuple` in normal code?
It makes reading the code much harder than it would otherwise be.

Reply:
I originally added `Tuple` to help with metaprogramming tasks, and I remain of the opinion that it shouldn't be used in non-generic code, and definitely not as part of the API of something.

Reply:
Do you have an alternative to it? Or should I use a struct?

Reply:
A struct is the best alternative, I think.

Reply:
Adjusted with https://github.com/SerenityOS/serenity/pull/7650/commits/005176e4328438256179690c191c15d16bd9ebcc